component/prometheus/remotewrite/remote_write.go

```diff
@@ -103,10 +112,10 @@ func New(o component.Options, c Arguments) (*Component, error) {
 				return 0, fmt.Errorf("%s has exited", o.ID)
 			}
 
-			localID := prometheus.GlobalRefMapping.GetLocalRefID(res.opts.ID, uint64(globalRef))
+			localID := ls.GetLocalRefID(res.opts.ID, uint64(globalRef))
```

Comment:
This `localID` may be for a previous configuration of the component and may no longer be valid. Is that okay?

Reply:
There might be an issue with queue manager and seriesLabels but I think to fix that would require some Prometheus changes. Worst case a series would drop until the wal watcher reread a segment with them. This would also be a bug in our current incarnation.

Reply:
True, that would already be a bug :) Maybe we could prevent it by adding a component 'incarnation' ID to the current component ID, so they don't reuse labels when config changes. Let's park this discussion for the future.